django_x509/tests/test_ca.py

```diff
@@ -681,3 +681,9 @@ def test_ca_common_name_length(self):
         message_dict = context_manager.exception.message_dict
         self.assertIn('common_name', message_dict)
         self.assertEqual(message_dict['common_name'][0], msg)
+
+    def test_ca_without_key_length_and_digest_algo(self):
+        with self.assertRaises(ValidationError) as context_manager:
+            self._create_ca(key_length=None, digest=None)
+        self.assertIn('key_length', context_manager.exception.error_dict)
+        self.assertIn('digest', context_manager.exception.error_dict)
```

Comment:
Here's how I suggest to rewrite this test:

```suggestion
def test_ca_without_key_length_and_digest_algo(self):
        try:
            self._create_ca(key_length='', digest='')
        except ValidationError as e:
            self.assertIn('key_length', e.error_dict)
            self.assertIn('digest', e.error_dict)
        except Exception as e:
            self.fail(f'Got exception: {e}')
        else:
            self.fail('ValidationError not raised as expected')
```

The first except block captures the current exception we are getting.
The last else block would capture the situation in which the validation error is not raised as expected and it's our defence line against future bugs.

Reply:
Got it